spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/strategies/HandleHealthCheckStep.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.server.deployer.strategies;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.cloud.skipper.domain.Release;
+import org.springframework.cloud.skipper.domain.Status;
+import org.springframework.cloud.skipper.domain.StatusCode;
+import org.springframework.cloud.skipper.server.deployer.ReleaseManager;
+import org.springframework.cloud.skipper.server.domain.AppDeployerData;
+import org.springframework.cloud.skipper.server.repository.AppDeployerDataRepository;
+import org.springframework.cloud.skipper.server.repository.DeployerRepository;
+import org.springframework.cloud.skipper.server.repository.ReleaseRepository;
+import org.springframework.context.event.EventListener;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Responsible for taking action based on the health of the latest deployed release. If
+ * healthy, then delete applications from the previous release. Otherwise delete the
+ * latest deployed release. Delegates to {@link DeleteStep}.
+ *
+ * @author Mark Pollack
+ * @author Ilayaperumal Gopinathan
+ */
+public class HandleHealthCheckStep {
+
+	private final Logger logger = LoggerFactory.getLogger(HandleHealthCheckStep.class);
+
+	private final ReleaseRepository releaseRepository;
+
+	private final DeployerRepository deployerRepository;
+
+	private final AppDeployerDataRepository appDeployerDataRepository;
+
+	private final DeleteStep deleteStep;
+
+	private final HealthCheckProperties healthCheckProperties;
+
+	private ReleaseManager releaseManager;
+
+	public HandleHealthCheckStep(ReleaseRepository releaseRepository,
+			DeployerRepository deployerRepository,
+			AppDeployerDataRepository appDeployerDataRepository,
+			DeleteStep deleteStep,
+			HealthCheckProperties healthCheckProperties) {
+		this.releaseRepository = releaseRepository;
+		this.deployerRepository = deployerRepository;
```

Comment:
deployerRepository is not used in this class and hence can be removed.

Reply:
great, splitting up the steps made the # of dep classes go down, a good thing.